third_party/move/move-vm/runtime/src/data_cache.rs

```diff
@@ -295,4 +295,18 @@ impl<'r> TransactionDataCache<'r> {
         let ty_layout = loader.type_to_type_layout(&ty)?;
         Ok(self.event_data.push((guid, seq_num, ty, ty_layout, val)))
     }
+
+    pub(crate) fn emitted_events(
+        &self,
+        guid: Vec<u8>,
+        ty: Type,
```

Comment:
Is this necessary?

Reply:
could we? I feel it is impossible since guid matches, ty must match too.

Reply:
actually let me push back again:
* The Move contract API makes this impossible, since it expects an event handle and therefore we have some bizarre invariant if we hit this
* The Move core code API has no expectations that the types are consistent and so we create a new application specific behavior by adding this check. I could rewrite the code quite a bit to be more advanced, but I'm not inclined to add the complexity